From 0e89e3fa8dca3f08b8d41f1d874fd7b650ebedd8 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:58 -0400
Subject: [virt] kvm/ppc: book3s hv - Don't crash host on unknown guest interrupt

Message-id: <1410545655-205645-130-git-send-email-dzickus@redhat.com>
Patchwork-id: 94248
O-Subject: [RHEL7 PATCH 129/626] KVM: PPC: Book3S HV: Don't crash host on unknown guest interrupt
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit f3271d4c9011bf5f36a65b9b5e7caaef4280686d
Author: Paul Mackerras <paulus@samba.org>
Date:   Fri Sep 20 14:52:41 2013 +1000

    KVM: PPC: Book3S HV: Don't crash host on unknown guest interrupt

    If we come out of a guest with an interrupt that we don't know about,
    instead of crashing the host with a BUG(), we now return to userspace
    with the exit reason set to KVM_EXIT_UNKNOWN and the trap vector in
    the hw.hardware_exit_reason field of the kvm_run structure, as is done
    on x86.  Note that run->exit_reason is already set to KVM_EXIT_UNKNOWN
    at the beginning of kvmppc_handle_exit().

    Signed-off-by: Paul Mackerras <paulus@samba.org>
    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_hv.c b/arch/powerpc/kvm/book3s_hv.c
index 31e99ab..59cf70f 100644
--- a/arch/powerpc/kvm/book3s_hv.c
+++ b/arch/powerpc/kvm/book3s_hv.c
@@ -709,8 +709,8 @@ static int kvmppc_handle_exit(struct kvm_run *run, struct kvm_vcpu *vcpu,
   printk(KERN_EMERG "trap=0x%x | pc=0x%lx | msr=0x%llx\n",
    vcpu->arch.trap, kvmppc_get_pc(vcpu),
    vcpu->arch.shregs.msr);
+  run->hw.hardware_exit_reason = vcpu->arch.trap;
   r = RESUME_HOST;
-  BUG();
   break;
  }
 
-- 
1.7.1